pandas/tseries/period.py

```diff
@@ -271,10 +271,13 @@ def _from_arraylike(cls, data, freq, tz):
 
     @classmethod
     def _simple_new(cls, values, name=None, freq=None, **kwargs):
-        if not getattr(values, 'dtype', None):
+
+        if not com.is_integer_dtype(values):
```

Comment:
I think that'll recursively error - `PeriodIndex.__new__` will call `_simple_new` with the floats. 

Although I'm not sure what's wrong with the current suggestion?